app/decorators/user_decorator.rb

```diff
@@ -92,7 +88,9 @@ def qrcode(otp_secret_key)
       digits: TwoFactorAuthenticatable::OTP_LENGTH,
       interval: IdentityConfig.store.totp_code_interval,
     }
-    url = ROTP::TOTP.new(otp_secret_key, options).provisioning_uri(email)
+    url = ROTP::TOTP.new(otp_secret_key, options).provisioning_uri(
+      user.confirmed_email_addresses.take&.email,
```

Comment:
would it make sense to use the last signed in email? or do we want to not rely on that timestamp anymore?

Reply:
Yeah, the nil-safe operator in the current implementation threw me off. I don't know in what case we _wouldn't_ have an email address (confirmed or otherwise) though, so it's probably fine to use the last signed in email

Reply:
Switched to last signed in email address in https://github.com/18F/identity-idp/pull/7876/commits/0159652ff7e55f6ec2f942f21bb3860408a5dc7a